modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -898,8 +898,12 @@ public CompletableFuture<BinaryRow> get(BinaryRowEx keyRow, @Nullable InternalTr
         }
 
         if (tx.isReadOnly()) {
-            return evaluateReadOnlyRecipientNode(partitionId(keyRow), tx.readTimestamp())
-                    .thenCompose(recipientNode -> get(keyRow, tx.readTimestamp(), tx.id(), tx.coordinatorId(), recipientNode));
+            HybridTimestamp readTimestamp = tx.readTimestamp();
+
+            assert readTimestamp != null : "Read timestamp is not set for read-only transaction";
```

Comment:
How is this possible?
According to the API, this should not happen and the checks should be somewhere earlier or in the tests.
I suggest removing this check, it looks redundant.